eth/filters/filter_system.go

```diff
@@ -478,10 +514,11 @@ func (es *EventSystem) eventLoop() {
 		es.pendingLogsSub.Unsubscribe()
 		es.pendingBlockSub.Unsubscribe()
 		es.chainSub.Unsubscribe()
+		es.reOrgSub.Unsubscribe()
 	}()
 
 	index := make(filterIndex)
-	for i := UnknownSubscription; i < LastIndexSubscription; i++ {
```

Comment:
Why do we remove LastIndexSubscription?

Reply:
we added a subscription type called reOrgSubscription at the end which has an index of 8, index of LastIndexSubscriptions is 7, so it wasn't created the channels. Had to include the reOrgSubscription.

Reply:
check the starting of this file where we initialize the types